src/test/java/test/dataprovider/issue2267/SampleTestCase.java

```diff
@@ -0,0 +1,23 @@
+package test.dataprovider.issue2267;
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+import test.SimpleBaseTest;
+
+@Listeners(CustomListener.class)
+public class SampleTestCase extends SimpleBaseTest {
```

Comment:
You don't need `CustomListener`. You can actually add the retry analyser directly via the `retryAnalyser` attribute of the `@Test` method.

Reply:
And samples don't need to extend SimpleBaseTest

Reply:
> You don't need `CustomListener`. You can actually add the retry analyzer directly via the `retryAnalyser` attribute of the `@Test` method.

But this actually wont trigger the bug that I wan't to fix. If you add the Analyzer via attribute it is already set as "default".

Reply:
@erickubenka - Got it! That makes sense.

Reply:
Can you just refactor `SimpleTestCase` to not extend `SimpleBaseTest` and push the changes. I think we should be good to get this merged. We are preparing for a release. Lets target to have this PR also as part of the upcoming release.